### PR TITLE
Adding GAC costing package documentation to unit_model doc

### PR DESCRIPTION
**Pull request by hunterbarber**
## Fixes/Resolves:
 Initial implementation of GAC costing package documentation

## Summary/Motivation:
Starting documentation for more complex costing functions

## Changes proposed in this PR:
Housed in unit model documentation until devoted costing package docs are constructed

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by hunterbarber** [comment]
@lbianchi-lbl @aladshaw3 I can't seem to link the [costing source code](https://watertap--637.org.readthedocs.build/en/637/apidoc/watertap.costing.html#watertap.costing.watertap_costing_package.WaterTAPCostingData.cost_gac), line 330, correctly. Seeing you both are the recent contributors to the how_to_contribute_development.rst, can you help me with understanding the issue or letting me know if it's possible.  Not sure what to see what type of class it should be given the auto generation.

**Comment by hunterbarber** [comment]
> @lbianchi-lbl @aladshaw3 I can't seem to link the [costing source code](https://watertap--637.org.readthedocs.build/en/637/apidoc/watertap.costing.html#watertap.costing.watertap_costing_package.WaterTAPCostingData.cost_gac), line 330, correctly. Seeing you both are the recent contributors to the how_to_contribute_development.rst, can you help me with understanding the issue or letting me know if it's possible. Not sure what to see what type of class it should be given the auto generation.

@andrewlee94 Can you take a look at the [previous commit using :meth:](https://github.com/watertap-org/watertap/pull/637/commits/21cfa3cbe525eb77e020db1ad7d89cefb8bf6b94) per your suggestion that it was the correct class and as to why it wasn't properly linking? Again for reference I'm trying to direct link to the [cost_gac](https://watertap--637.org.readthedocs.build/en/637/apidoc/watertap.costing.html#watertap.costing.watertap_costing_package.WaterTAPCostingData.cost_gac), one level below how its linked in the [current commit](https://github.com/watertap-org/watertap/pull/637/commits/c9d10ebe045feb7450248ec9a643b2c43e098318).

**Comment by lbianchi-lbl** [comment]
@hunterbarber I've tried to check out commit 21cfa3cbe525eb77e020db1ad7d89cefb8bf6b94 and build the docs locally, and it seems to work fine:

image
image

I think the syntax in that commit should be the correct one:

```rst
:meth:`watertap.costing.watertap_costing_package.WaterTAPCostingData.cost_gac`
```

Can you try to re-apply that change, push again to this PR, and check that the link is correct in the resulting fresh ReadTheDocs build?

File: docs/technical_reference/unit_models/gac.rst [hunk 1]
```diff
@@ -237,15 +237,105 @@ if ``surface_diffusion_coefficient_type`` is set to ``calculated``:
...
+
+        & C_{op,tot} = C_{op,regen}+C_{op,makeup} \\\\
+        & C_{op,regen} = f_{regen}C_{regen}\dot{m}_{GAC} \\\\
+        & C_{op,makeup} = \left( 1-f_{regen} \right)C_{makeup}\dot{m}_{GAC}
+
+Code Documentation
 -------------------
 
 * :mod:`watertap.unit_models.gac`
+* :func:`watertap.costing.watertap_costing_package.WaterTAPCostingData`
```
Comment (lbianchi-lbl) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
* :meth:`watertap.costing.watertap_costing_package.WaterTAPCostingData.cost_gac`
```

Comment (lbianchi-lbl) [reply docs/technical_reference/unit_models/gac.rst]: @hunterbarber You should be able to just click on "Commit suggestion" to trigger a new ReadTheDocs build.

**Comment by hunterbarber** [comment]
> Can you try to re-apply that change, push again to this PR, and check that the link is correct in the resulting fresh ReadTheDocs build?

@lbianchi-lbl You're definitely right, it must've just been an oversight by me on this PR when I was checking the several different options. Thanks for taking a look.

**Review by TimBartholomew — APPROVED** [review]
LGTM - I didn't check all the equations, but I like the formatting and level of detail

**Review by adam-a-a — APPROVED** [review]
LGTM - we'll probably migrate at some point once we get watertap costing package documentation in place, so we can always revise further at that point, _if needed_.

**Comment by TimBartholomew** [comment]
Yeah we should figure out where we want costing documentation

**Pull request merged**